core/src/proof.rs

```diff
@@ -100,6 +100,45 @@ pub struct VerifiedPathProof {
     terminal: Option<LeafData>,
 }
 
+impl VerifiedPathProof {
+    /// Get the terminal node.
+    pub fn terminal(&self) -> Option<&LeafData> {
+        self.terminal.as_ref()
+    }
+
+    /// Get the proven path.
+    pub fn path(&self) -> &BitSlice<u8, Msb0> {
+        &self.key_path.view_bits()[..self.depth as usize]
```

Comment:
Maybe warrants a comment on `depth` declaration that it's length ≤ key_path.len()

Reply:
That's protected by the type `u8`